tools/buildmgr/cbuildgen/src/CMakeListsGenerator.cpp

```diff
@@ -46,7 +46,15 @@ bool CMakeListsGenerator::GenBuildCMakeLists(void) {
   if (!m_asMscGlobal.empty())        cmakelists << EOL << "set(AS_FLAGS_GLOBAL \"" << CbuildUtils::EscapeQuotes(m_asMscGlobal) << "\")";
   if (!m_ccMscGlobal.empty())        cmakelists << EOL << "set(CC_FLAGS_GLOBAL \"" << CbuildUtils::EscapeQuotes(m_ccMscGlobal) << "\")";
   if (!m_cxxMscGlobal.empty())       cmakelists << EOL << "set(CXX_FLAGS_GLOBAL \"" << CbuildUtils::EscapeQuotes(m_cxxMscGlobal) << "\")";
-  if (!m_linkerMscGlobal.empty())    cmakelists << EOL << "set(LD_FLAGS_GLOBAL \"" << CbuildUtils::EscapeQuotes(m_linkerMscGlobal) << "\")";
+  if (!m_linkerMscGlobal.empty()) {
+    cmakelists << EOL << "set(LD_FLAGS_GLOBAL \"" << CbuildUtils::EscapeQuotes(m_linkerMscGlobal);
+    if(!m_cxxFilesList.empty()) {
+      if (!m_linkerCxxMscGlobal.empty()) cmakelists << " " << CbuildUtils::EscapeQuotes(m_linkerCxxMscGlobal);
+    } else {
+      if (!m_linkerCMscGlobal.empty()) cmakelists << " " << CbuildUtils::EscapeQuotes(m_linkerCMscGlobal);
+    }
+    cmakelists << "\")";
+  }
```

Comment:
This implementation is not correct. Please see my comment https://github.com/Open-CMSIS-Pack/devtools/pull/492#discussion_r976411127